File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # initialize hashmap
        hashmap = {}
        #iterate through array...len() calculates the length of the list and range() is like i++
        for i in range(len(nums)):
            complement = target - nums[i]
            if complement in hashmap:
                return [i, hashmap[complement]]
            hashmap[nums[i]] = i
        # Return an empty list if no solution is found
        return []

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) { 

    //Initialize hashmap
    let hashmap = new Map()
    //Iterate through array and add complements to hashmap
    for (let i = 0; i <= nums.length; i++) {
        //Initialize complement and define
        let complement = target - nums[i]
        //Check if complement is in hashmap and if true, return answer
        if (hashmap.has(complement)) {
            return [i, hashmap.get(complement)]
        }
        //If false, complement is not present, add to hashmap
        hashmap.set(nums[i], i)
    }
};